### ISSUE TITLE
How to run Compass under native Wayland?

### ISSUE THREAD

**Issue by sepehr** [mongodb-js/compass#3938]
Since Electron 12.x, there is support for native Wayland through the use of these feature flags:

```
--enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform=wayland
```

or using `--ozone-platform-hint=auto`. This is the way to avoid a badly-stretched blurry, and hardly-usable UI on HiDPI displays. HiDPI Wayland users usually run VSCode, Chromium itself, and other popular Electron apps like so.

Passing those arguments to Compass on a Fedora 37, Gnome 4.32 desktop using:

```
flatpak run com.mongodb.Compass --enable-features=UseOzonePlatform --ozone-platform=wayland
```

Compass crashes with:

```
[3:0105/181354.163494:ERROR:wayland_connection.cc(180)] Failed to connect to Wayland display
[3:0105/181354.163517:FATAL:ozone_platform_wayland.cc(211)] Failed to initialize Wayland platform
/app/bin/run.sh: line 2:     3 Trace/breakpoint trap   (core dumped) zypak-wrapper '/app/main/MongoDB Compass' "$@"
```

I wonder how we could safely pass native wayland feature flags down to Electron and Chromium? It's definitely not a pleasure to use Compass in XWayland. Passing `--ignore-additional-command-line-flags` did not help.

---

![blurry compass under XWayland](https://user-images.githubusercontent.com/23576/210839776-8d9b28bc-61d5-473d-b435-f116a7f61584.png)

**Comment by addaleax** [comment]
@sepehr We aren’t doing that much in terms of fancy work on top of Electron, so this _probably_ isn’t very Compass-specific. At the same time, I’m not an expert for how Electron/Chromium work with different display servers.

Do you know if there would be a way to repro this without having to fully set up a physical machine?

**Comment by sepehr** [comment]
@addaleax Thanks for the follow-up and apologies for the my late reply. 

I have to look closer tbh. However, knowing that there is upstream support for native Wayland in Electron, Compass job is done if we are sure that the arguments it does not recognize are passed downwards to Electron rather than being omitted.

Here's how VSCode output look like when it's passed the wayland-specific arguemnts: 

```
Warning: 'enable-features' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'ozone-platform' is not in the list of known options, but still passed to Electron/Chromium.
```

I hope I'll find the time to look into this closer.

**Comment by ahmafi** [comment]
I ended up using the `rpm` package instead of flatpak. Like this:
```sh
mongodb-compass --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform=wayland --ignore-additional-command-line-flags
```

**Comment by sepehr** [comment]
Thanks for your input @ahmafi. I ended up ditching fractional scaling to address my issues. With 200% scaling, things are vivid again.

**Comment by seantcanavan** [comment]
the `flatpak` route did not help me unfortunately as I just got different critical GPU errors:

`flatpak run com.mongodb.Compass --enable-features=UseOzonePlatform --ozone-platform=wayland`

However, this approach finally did work but after a reboot. I suspect I had too many pre-existing versions of the app already run with different parameters after trying a bunch of different things to get it to start up.

`yay -Syu mongodb-compass
mongodb-compass  --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform=wayland --ignore-additional-command-line-flags`

**Comment by Ugzuzg** [comment]
It would be great if the app also respected `$XDG_CONFIG_HOME/electron-flags.conf` options.

**Comment by faileon** [comment]
I am gonna post my fair share of problems with wayland and mongo-compass, the whole discussion is here:
https://bbs.archlinux.org/viewtopic.php?pid=2198925#p2198925

tl;dr is that if I have env variable
`ELECTRON_OZONE_PLATFORM_HINT`
set to any value `auto` or `wayland`, the app will not render at all. I must unset the variable to get compass to render. This will use xwayland, but at least it works.

In combination with fixing credential storage, my desktop file
`/usr/share/applications/mongodb-compass.desktop`
by setting the Exec line to:
`Exec=env MONGODB_COMPASS_TEST_LOG_DIR=/dev/null ELECTRON_OZONE_PLATFORM_HINT= mongodb-compass --ignore-additional-command-line-flags --password-store="gnome-libsecret" %U`

**Comment by programista-zacny** [comment]
> It would be great if the app also respected `$XDG_CONFIG_HOME/electron-flags.conf` options.

It would be great! Like `chrome-flags.conf`, `chromium-flags.conf` etc.
Maybe dedicated `mongodb-compass-flags.conf`?
